### PR TITLE
fix issue 29191: only try partial matches when a wildcard has been sp…

### PR DESCRIPTION
**Pull request by bastiaanb**
…ecified

### CODE REVIEW

File: tests/unit/utils/utils_test.py [hunk 1]
```diff
@@ -264,6 +265,24 @@ def test_subdict_match(self):
                 test_two_level_dict_and_list, 'abc:lorem:ipsum:dolor:sit'
             )
         )
+        # Test four level dict match for reference
```
Comment (cachedout) [inline tests/unit/utils/utils_test.py]: Could you please remove the trailing whitespace here?

**Comment by cachedout** [comment]
@bastiaanb Thanks for this. There is one small lint issue that I commented on in-line and then it looks like this change is causing some test failures. Could you please have a look?

**Comment by bastiaanb** [comment]
@cachedout : updated the unit tests. Since the original feature did not include test cases I'm a bit unsure about the intended behaviour of the wildcard matching, so I focus on fixing the regression only.

**Comment by cachedout** [comment]
@bastiaanb OK, sounds good. It will take me a little while to carefully review this but we'll try and get this in as soon as possible. Thanks for your hard work here!

**Comment by cachedout** [comment]
Looks good. Merged. Thanks @bastiaanb

**Comment by jfindlay** [comment]
Fixes #29191.

**Comment by jfindlay** [comment]
This needs to be backported into 2015.8

**Pull request merged**